plugins/trace_api_plugin/request_handler.cpp

```diff
@@ -26,105 +26,168 @@ namespace {
-               ("data", fc::to_hex(a.data.data(), a.data.size()));
-
-         auto params = data_handler(a, yield);
+         auto action_variant =  e_action_trace ==  e_action_trace_v0?   fc::mutable_variant_object()
+               ("global_sequence", std::get<action_trace_v0>(a).global_sequence)
+               ("receiver", std::get<action_trace_v0>(a).receiver.to_string())
+               ("account", std::get<action_trace_v0>(a).account.to_string())
+               ("action", std::get<action_trace_v0>(a).action.to_string())
+               ("authorization", process_authorizations(std::get<action_trace_v0>(a).authorization, yield))
+               ("data", fc::to_hex(std::get<action_trace_v0>(a).data.data(), std::get<action_trace_v0>(a).data.size())) :
+            e_action_trace == e_action_trace_v1 ?  fc::mutable_variant_object()
+               ("global_sequence", std::get<action_trace_v1>(a).global_sequence)
+               ("receiver", std::get<action_trace_v1>(a).receiver.to_string())
+               ("account", std::get<action_trace_v1>(a).account.to_string())
+               ("action", std::get<action_trace_v1>(a).action.to_string())
+               ("authorization", process_authorizations(std::get<action_trace_v1>(a).authorization, yield))
+               ("data", fc::to_hex(std::get<action_trace_v1>(a).data.data(), std::get<action_trace_v1>(a).data.size())) 
+               ("return_value", fc::to_hex(std::get<action_trace_v1>(a).return_value.data(), std::get<action_trace_v1>(a).return_value.size())) : fc::mutable_variant_object();
+
+         auto params = e_action_trace == e_action_trace_v0 ? data_handler_v0(std::get<action_trace_v0>(a), yield) : data_handler_v1(std::get<action_trace_v1>(a), yield) ;
          if (!params.is_null()) {
             action_variant("params", params);
          }
+         if(e_action_trace == e_action_trace_v1){
+            auto return_data = return_handler(std::get<action_trace_v1>(a), yield);
+            if(!return_data.is_null()){
+               action_variant("return_data", return_data);
+            }
+         }
 
          result.emplace_back( std::move(action_variant) );
+
       }
 
       return result;
 
    }
 
-   fc::variants process_transactions(const std::vector<transaction_trace_v0>& transactions, const data_handler_function& data_handler, const yield_function& yield ) {
+  
+   template<typename action_trace>
+   vec_action_trace_variant to_action_trace_variant(const std::vector<action_trace> & actions){
+       vec_action_trace_variant vv;
+	for (auto &  action : actions){
+		vv.push_back(action);
+	}
+	return vv;
+   }
+
+   fc::variants process_transactions(trace_api_type_enum e_transaction_trace, vec_transaction_trace_variant && transactions, const data_handler_function_v0& data_handler_v0, const data_handler_function_v1 & data_handler_v1,  const return_handler_function & return_handler, const yield_function& yield ) {
```

Comment:
Same as above with utilizing templates an `if constexpr`.

Reply:
It's a good idea.

Reply:
Combined using template, same as above.